pkg/node/node.go

```diff
@@ -454,27 +455,63 @@ func NewBee(interrupt chan struct{}, addr string, publicKey *ecdsa.PublicKey, si
 		return nil, err
 	}
 
-	var (
-		blockHash []byte
-		txHash    []byte
-	)
-
-	txHash, err = GetTxHash(stateStore, logger, o.Transaction)
+	// if theres a previous transaction hash, and not a new chequebook deployment on a node starting from scratch
+	// get old overlay
+	// mine nonce that gives similar new overlay
+	nonce, nonceExists, err := overlayNonceExists(stateStore)
 	if err != nil {
-		return nil, fmt.Errorf("invalid transaction hash: %w", err)
+		return nil, fmt.Errorf("check presence of nonce: %w", err)
+	}
+	if !nonceExists {
+		nonce = make([]byte, 32)
 	}
 
-	blockHash, err = GetTxNextBlock(p2pCtx, logger, chainBackend, transactionMonitor, pollingInterval, txHash, o.BlockHash)
+	existingOverlay, err := GetExistingOverlay(stateStore)
 	if err != nil {
-		return nil, fmt.Errorf("invalid block hash: %w", err)
+		logger.Warning("existing overlay", "error", err)
+	}
+
+	if err == nil && o.FullNodeMode && !nonceExists {
```

Comment:
no need to check err

Reply:
this err == nil is needed, because if we dont find existing overlay in store, ErrNotFound is not suppressed, but also, we have no existing overlay to mine a similar new overlay to, so in this case we just going to use the zero nonce as any overlay would be fine